### PR TITLE
FAQ snippets

### PR DESCRIPTION
**Pull request by devinmarieb**
# Description
- Create a new snippets/faq directory to start the process of organizing all FAQs into snippets
- Created snippets for features/pages with existing integrated FAQs
- Standardized the styling of FAQs
- Removes the CircleCI Academy FAQ page from side nav as this was meant to be a temporary page (will not archive quite yet as there is no clear solution where to redirect this URL to as we have no other pages dealing with CCI Academy - another solution is to create a small page for this resource).

This PR does not touch the styling or integration of anything on the main FAQ page.

This PR also does not yet address the mixing of some troubleshooting questions in FAQ sections/pages.

# Reasons
We have an inconsistent way of keeping track of feature FAQs. Some features have FAQS mixed in existing pages, some features have their own separate FAQ pages, and there is a large FAQ page that is incomplete. This starts to work toward aggregating all information in one place, in one standard style.

Using snippets for all FAQs will allow us to use the same info in the same style in multiple places, with the overall goal being that all features have an independent, dedicated FAQ page in addition to all the information living in one giant main FAQ page (similar to the config ref).

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_includes/snippets/faq/self-hosted-runner-faq-snip.adoc [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+=== What are the best practices for managing state between jobs?
+
+Machine runners are unopinionated about state between jobs. Machine runners can be configured to give each job a unique working directory and clean it up afterwards - but this is optional. And by default, nothing restricts the job from placing files outside of its working directory.
+
+In general CircleCI recommends jobs rely on as little state as possible to improve their reproducibility. An effective way to accomplish this is to put cleanup steps at the start of a job so they are guaranteed to run regardless of what happened to a previous job.
+
+CAUTION: It may be possible to reduce build times by making use of caches that persist on the host between jobs, however this is a trade-off against reproducibility - and may also lead to disks filling up over time. As a result, this trade-off could lead to higher billable usage.
+
+[#how-long-do-inactive-self-hosted-runners-persist-in-the-self-hosted-runner-inventory-page]
+==== How long do inactive self-hosted runners persist in the self-hosted runner inventory page?
```
Comment (rosieyohannan) [inline jekyll/_includes/snippets/faq/self-hosted-runner-faq-snip.adoc:69]: Just checking if it's right that this is an h4 rather than h3 like the others?

**Review by rosieyohannan — COMMENTED** [review]
Looks great! Just one little question :-D

File: jekyll/_includes/snippets/faq/self-hosted-runner-faq-snip.adoc (hunk 1 above)
Comment (devinmarieb) [reply jekyll/_includes/snippets/faq/self-hosted-runner-faq-snip.adoc]: @rosieyohannan all fixed!

**Review by rosieyohannan — APPROVED** [review]
🎉

**Pull request merged**